### ISSUE TITLE
Randomly Not Able to Receive or Transmit on 152/117

### ISSUE THREAD

**Issue by Contendify** [IDI-Systems/acre2#1079]
ARMA 3 Latest Build
CBA Latest Build
ACRE2 Latest Build
ACE Latest Build

**Description:**

Start a mission ranging from 5-25 people having 0 radio issues for usually the first 20-30 minutes only to have radios randomly not able to send or receive. Sometimes you will be able to send and maybe someone will receive it but when they try to respond they cant send and you cant hear them. We use 152s/117s and have an issue with both. We will be fine in one position and without moving it will start to not work.

**Steps to reproduce:**
Currently, no way to reproduce since there is no sure sign of why it happens. 

**Expected behavior:**
Have radios work for the entirety of Op.

**Where did the issue occur?**
- Dedicated 

**Log Files:**
Sadly no logs since -nologs. Nothing pops in the .rpt anyway.

I have seen an issue like this happening in 2020 and some people said it was fixed with the newest release. Sadly may have to switch to TFAR for this issue but REALLY REALLY don't want to.

**Comment by jonpas** [comment]
Do you run any scripts that change inventories? Maybe there is some action you perform that causes this.

**Comment by Contendify** [comment]
> Do you run any scripts that change inventories? Maybe there is some action you perform that causes this.

What script may you be referring to? The only thing that we do use that sounds anything like that is the Backpack on Chest Mod.

**Comment by jonpas** [comment]
Any mission scripts or any mod. Backpack on chest is probably not causing it.

Do you run custom radio presets by any chance?

**Comment by Contendify** [comment]
> Any mission scripts or any mod. Backpack on chest is probably not causing it.
> 
> Do you run custom radio presets by any chance?

Yes we do run a custom radio preset. Although I have already tested it with and without the radio preset. The only fix that we have came accross that is a temporary fix is the turning the terrain coef down to 0 which sucks. I will link my custom radio preset below. 

https://pastebin.com/FULUrLCy

**Comment by jonpas** [comment]
Can you try running a mission or a few without custom presets? There is issue #1056 which has a hint that using custom preset names might break something (I still don't get how that would be, but that's all the evidence so far). You don't actually use the custom-named preset so your `copyPreset` function calls are redundant. But maybe that tells us the issue is deeper in presets. That's why I suggest running without for a few missions and see if you still get those issues.

Also mind, make sure issue is an actual issue and not an actual radio signal loss (remember, radios suck realistically). :D

**Comment by Contendify** [comment]
Yeah, it 100% is an actual issue. I've used ACRE for years and know when shit isn't working due to terrain loss etc. I am gonna keep trying the next couple of weeks' mission without the preset. Is there a good preset that I can just change the label names to my own? I also very much appreciate the help, my man. Part of my unit wants to go back to TFAR and I am doing everything in my power not to.

**Comment by jonpas** [comment]
I would suggest no preset at all, as just changing the name seemed to have caused issues.

Tell them to run ACRE with "Arcade" mode. :joy:

**Comment by JordannDev** [comment]
I've just run into this issue tonight with my group while testing. Same problem. We played for around 3 hours, and randomly near the end, the radios stopped working.

No custom preset, but I will note in the beginning I did change the name of the second channel, and I also swapped the frequency. Even after I died, with the frequency set correctly still not able to hear anything.

Other players were able to hear people transmitting at some point. We ran into this issue a long while ago when we were testing ACRE.

**Comment by Contendify** [comment]
> I've just run into this issue tonight with my group while testing. Same problem. We played for around 3 hours, and randomly near the end, the radios stopped working.
> 
> No custom preset, but I will note in the beginning I did change the name of the second channel, and I also swapped the frequency. Even after I died, with the frequency set correctly still not able to hear anything.
> 
> Other players were able to hear people transmitting at some point. We ran into this issue a long while ago when we were testing ACRE.

Are you using Real Engine by chance? I thought it might have been Real Engine so we removed it. Sadly the only way to know if that is it is by trying to break it during an op :(

**Comment by JordannDev** [comment]
> > I've just run into this issue tonight with my group while testing. Same problem. We played for around 3 hours, and randomly near the end, the radios stopped working.
> > No custom preset, but I will note in the beginning I did change the name of the second channel, and I also swapped the frequency. Even after I died, with the frequency set correctly still not able to hear anything.
> > Other players were able to hear people transmitting at some point. We ran into this issue a long while ago when we were testing ACRE.
> 
> Are you using Real Engine by chance? I thought it might have been Real Engine so we removed it. Sadly the only way to know if that is it is by trying to break it during an op :(

Nope, we were not using it. I'd be happy to provide a list of mods we are using if needed.

**Comment by Kofeina101** [comment]
My group also encountered this issue few times afer few hours of mission, also sometimes after similar time ACRE is restarting(looks like it) every few seconds, i can see that by looking at racks, it blinks as its switching on and off constantly, it also breaks intercom. 
All mods from Workshop- ACE, ACRE, CBA, RHS, CUP. 
We are also using dedicated server. 
No radio presets. 
Couldn't find anything in server logs. 
No scripts involving inventory. 
I wonder if issue is bc of some dedi server settings, too low FPS or i am doing something wrong

Recently i switched to Longley-Rice, and for now it works okey, but we didn't play any +3hrs long op so it need proper testing.

**Comment by Zumiii** [comment]
For us, this exact same problem exists to. We are using similar mods. The problem seems to rise when 15+ players are present. We are playing on Reshmaan and are using SEM 70.
Radio IDs are properly unique, but sometimes players standing next to each other will just not receive the same transmissions.
If it helps, I have added Radio-Racks to Vehicles on the server:
```
[_entity, ["ACRE_SEM90", "Zugfunkkreis", "ZgKrs", false, ["gunner"], [], "ACRE_SEM70", [], ["intercom_1"]], true, {}] call acre_api_fnc_addRackToVehicle;
[_entity, ["ACRE_SEM90", "Gruppenfunkkreis", "GrpKrs", false, ["gunner"], [], "ACRE_SEM70", [], ["intercom_1"]], true, {}] call acre_api_fnc_addRackToVehicle;
  
```

As far as I read, this might cause it? Any updates on this?
I might change our signal calculation to Longley-Rice as well for a temporary fix.

**Comment by JordannDev** [comment]
> My group also encountered this issue few times afer few hours of mission, also sometimes after similar time ACRE is restarting(looks like it) every few seconds, i can see that by looking at racks, it blinks as its switching on and off constantly, it also breaks intercom.
> All mods from Workshop- ACE, ACRE, CBA, RHS, CUP.
> We are also using dedicated server.
> No radio presets.
> Couldn't find anything in server logs.
> No scripts involving inventory.
> I wonder if issue is bc of some dedi server settings, too low FPS or i am doing something wrong
> 
> Recently i switched to Longley-Rice, and for now it works okey, but we didn't play any +3hrs long op so it need proper testing.

We are also using similar mods. We didn't use many (if any) vehicles in our scenario though.

**Comment by LazyBoot** [comment]
We've been running into issues like this quite a lot recently as well
running a fairly minimal modset on the new SOG Mike Force mission
radios stopped working in less than 30 minutes
![image](https://user-images.githubusercontent.com/3827482/118409640-c1056e00-b68b-11eb-9425-ada080010907.png)

Edited to add: at some points we couldn't even hear each other transmit while in the same vehicle, which should rule out any terrain issues.

**Comment by JordannDev** [comment]
> We've been running into issues like this quite a lot recently as well
> running a fairly minimal modset on the new SOG Mike Force mission
> radios stopped working in less than 30 minutes
> ![image](https://user-images.githubusercontent.com/3827482/118409640-c1056e00-b68b-11eb-9425-ada080010907.png)
> 
> Edited to add: at some points we couldn't even hear each other transmit while in the same vehicle, which should rule out any terrain issues.

Our group did some stuff on the new SOG missions as well, we used similar mods. We ran into the problem again, but I will note, once we switched to Longley-Rice and it worked without any issues (do note though, we were only in-game for an hour).

Mods used:
![image](https://user-images.githubusercontent.com/26309778/118734089-a22de580-b80b-11eb-86b3-44d1b171e2d2.png)

**Comment by LazyBoot** [comment]
> Our group did some stuff on the new SOG missions as well, we used similar mods. We ran into the problem again, but I will note, once we switched to Longley-Rice and it worked without any issues (do note though, we were only in-game for an hour).

Are there any downsides to using Longley-Rice compared to the default?

**Comment by Schmitt-DE** [comment]
4 weeks in a row we played the same mission with about 15-18 players using modified ACRE presets. Always multiple transmission errors during the mission: Someone could not hear the radios. Someone could not send anything over the radio.

Now I disabled the modified presets and got back to default presets => We tried the same mission (just with deactivated presets) yesterday with 23 players for 4 hours and everything runs fine.

The killer function was: acre_api_fnc_setPresetChannelField

Not using acre_api_fnc_setPresetChannelField anymore => No more ACRE bugs :-)

But also no more preset channel frequencies :-( Now everyone has to configure his radio manually again at the beginning of the mission :-(
